modules/gdscript/language_server/gdscript_language_protocol.h

```diff
@@ -31,33 +31,55 @@
 #ifndef GDSCRIPT_PROTOCAL_SERVER_H
 #define GDSCRIPT_PROTOCAL_SERVER_H
 
+#include "core/io/stream_peer.h"
+#include "core/io/stream_peer_tcp.h"
+#include "core/io/tcp_server.h"
 #include "gdscript_text_document.h"
 #include "gdscript_workspace.h"
 #include "lsp.hpp"
 #include "modules/jsonrpc/jsonrpc.h"
-#include "modules/websocket/websocket_peer.h"
-#include "modules/websocket/websocket_server.h"
+
+#define LSP_MAX_BUFFER_SIZE 4096
```

Comment:
I think the buffer size is too small.
You can try open a script large than 4KB

Reply:
I would also suggest to use `constexpr` variable instead of macro.

Reply:
@Shatur95 To my knowledge, we don't allow the use of `constexpr` in Godot's codebase. We currently only allow C++03 code to be merged; a subset of C++11 (or C++14) will soon be allowed.

Reply:
@Calinou, bad to hear :( What is the reason of it?

Reply:
@Shatur95 The reason we only plan on allowing a subset of modern C++ is to ease online code reviews (among other things). When you're a pull request reviewer, you typically don't clone people's branches in an IDE; you just rely on GitHub's code viewer which doesn't provide things like type information on hover. For instance, this is why we'll disallow the use of `auto` in Godot's codebase.

Reply:
@Calinou, this decision is strange to me. I agree that using `auto` can really reduce readability. But this is because it does not intended to be used always. For example, in the constructions such `auto * widget = new QWidget` looks good because allows to get rid of duplicate type specifying.
`constexpr` as for me is always better than macro constant because it do the same but in safe way.

Reply:
We will use `constexpr` in the future, but this codebase is currently C++03 and introducing a stray `constexpr` when we have hundreds of `#define`s doesn't make sense. The C++14 port has to happen first.

Reply:
@akien-mga, good point, thanks for the explanation.

Reply:
@Geequlim any recomendations about a reasonable buffer size?  I agree that 4kb is too small.